rfcs/rfc-0000-interactions.md

```diff
@@ -0,0 +1,599 @@
+    },
+    Member {
+        member: Member,
+        user: User,
+    },
+    Channel {
+        channel: Channel,
+    },
+    Role {
+        role: Role,
+    },
+    Date {
+        value: String,
+    },
+    Time {
+        value: u32,
+    },
+    Datetime {
+        date: String,
+        time: u32,
+    },
+    Attachment {
+        value: String,
+    },
+    VarArgs {
+        values: Vec<InteractionDataSlashCommandOptionInner>,
+    },
+    Group {
+        group: InteractionDataSlashCommandOption,
+    },
+}
+```
+
+## Responding To Interactions
+
+After receiving an interaction the bot has 3 seconds to respond to the interaction with a reponse, this
+response can be either a message if the interaction is attached to a channel or a modal.
+
+As not all validation can happen client side and api side the response can be an error response to indicate
+something failed validation or some other error for the user, this can include a message to the user to
+indicate why it failed, it can also optionally reenter the data they entered to make it easier to change the
+data the user typed in and send correct data.
+
+If the interaction is attached to a channel the interaction will also include a webhook which can be used
+to send followup messages, this webhook will last for 15 minuites then expire.
+
+## Config Panel
+
+The config panel structure is fetched everytime it is accessed, unlike slash commands where they must be synced,
+this is because it can include a lot of dynamic data like defaults - which should be the existing config values
+if any, select choices which use any members, roles, channels or similar from the context of where the panel is
+being used.
+
+There will be a single save button which initiats the interaction flow for the config panel, like other
+interaction types, this can lead into new interactions like modals, this can also respond with a failed validation
+response indicating that the values where wrong or some other error.
+
+There are two interaction flows present in config panels:
+
+- A user opens the panel, the API then must initiate a interaction flow to fetch the config panel, this is then
+displayed on the client to the user.
+
+- A user edits and saves the values in the config panel, this initiate a interaction flow to forward the config
+values to the bot to be saved.
+
+## Dashboard
+
+todo
+
+## Routes
+
+## Slash Command Management
+
+These routes are for managing slash commands on the bot
+
+```http
+GET /interactions/slash_commands - Get interactions
+PUT - /interactions/slash_commands - Sync interaction
+```
+
+## Config Panel
+
+
+These routes are the server and channel bot config panels
+
+```http
+GET /interactions/config_panel - Get server/channel config - spawns interactions
+PATCH /interactions/config_panel - Update server/channel config
+PATCH /interactions/config_panel/invoke - Set server/channel config options - spawns interactions
+```
+
+## Responding To Interactions
```

Comment:
Not sure if these headers are intentional, but shouldn't they be `h3`?
Given the text descriptions below them and the empty `## Routes` header I believe they should be part of said header and therefore be `h3`.